### PR TITLE
Upgrade

### PR DESCRIPTION
**Pull request by dcoxall**
@joshuafleck @calo81 @ngsmrk 

To help fight the good fight against ET I wanted to make some changes to the client our apps use. Imagine my surprise to find it:

- Out-of-date with their own
- Development dependencies locked to only variants of major version 0 (rspec < 1 for instance)

So what did I do?

I've merged their latest version of master into this, accounting for changes we have also made.

BUT of course it's not that simple is it? Tests fail, a lot fail in-fact. So ...

1. I have upgraded dependencies
2. Fixed specs and many warning messages
3. Corrected code that caused failing tests

At this point everything is now passing! A green build! Yet another BUT however is some of the code that I added to correct the failing tests was already there... commented out... and it had been commented out by them roughly 2 years ago (https://github.com/simplybusiness/FuelSDK-Ruby/commit/4bdd4c9201f74934cedfff875685e0a4326aecd8#diff-6f9fd668b6856e0f269e884ab514bbf6R202) but I have no way of knowing if I have introduced a bug and what the tests should actually be or if I have fixed a bug and the tests are correct.

At this point they have also added a license and so not sure what our stance is on the licensing.

**I have not yet tested this with any of our apps**

### CODE REVIEW

**Comment by dcoxall** [comment]
I'd like to add the tabs are them, not me haha

**Comment by calo81** [comment]
@dcoxall Elliott uses currently a different branch than master of this Gem: https://github.com/ExactTarget/FuelSDK-Ruby/compare/master...simplybusiness:rest_adapt_to_new_savon?w=1

There we incompatibility problems in the original SOAP message formats that made the API calls not work.. And it seemed that using this version then Chopin didn't quite work at the time , so the 2 branches are kept..... 

Please review that branch and feel free to retest Chopin with it and with the new included changes you are doing to see if it works

File: fuelsdk.gemspec [hunk 1]
```diff
@@ -4,25 +4,25 @@ $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
...
+	spec.test_files    = spec.files.grep(%r{^(samples|test|spec|features)/})
+	spec.require_paths = ["lib"]
 
-  spec.add_development_dependency "bundler", "~> 1.3"
-  spec.add_development_dependency "rake"
-  spec.add_development_dependency "rspec"
-  spec.add_development_dependency "guard"
-  spec.add_development_dependency "guard-rspec"
+	spec.add_development_dependency "bundler", "~> 1.3"
+	spec.add_development_dependency 'rake'
```
Comment (joshuafleck) [inline fuelsdk.gemspec]: consistency with `'` or `"` quotes

**Comment by joshuafleck** [comment]
assuming this passes extensive manual QA, I'm fine with this :+1: 

signed-off-by: @joshuafleck

**Comment by amar-shah** [comment]
Closing down we can re-open new upgrade when required.

**Pull request closed without merging**